checker/src/main/java/org/checkerframework/checker/signature/SignatureAnnotatedTypeFactory.java

```diff
@@ -132,20 +130,52 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
                         && arg1.getKind() == Tree.Kind.STRING_LITERAL) {
                     String const0 = (String) ((LiteralTree) arg0).getValue();
                     String const1 = (String) ((LiteralTree) arg1).getValue();
-                    ExpressionTree receiver = TreeUtils.getReceiverTree(tree);
-                    final AnnotatedTypeMirror receiverType = getAnnotatedType(receiver);
-                    if (const0.equals(".")
-                            && const1.equals("/")
-                            && receiverType.getAnnotation(BinaryName.class) != null) {
-                        type.replaceAnnotation(INTERNAL_FORM);
-                    } else if (const0.equals("/")
-                            && const1.equals(".")
-                            && receiverType.getAnnotation(InternalForm.class) != null) {
-                        type.replaceAnnotation(BINARY_NAME);
+                    if (const0.length() == 1 && const1.length() == 1) {
+                        c1 = const0.charAt(0);
+                        c2 = const1.charAt(0);
                     }
                 }
+            } else {
+                return super.visitMethodInvocation(tree, type);
```

Comment:
A call to `super.visitMethodInvocation` should be the last invocation.
As you have changed it, this method does part of its work, then the superclass does its work, then this class does some more.  That makes the control flow more confusing and might lead to inconsistencies in the future.
(Trying to reduce code duplication is good!  But please don't move the super call from the end of the method.)